src/ship.js

```diff
@@ -0,0 +1,28 @@
+const Itinerary = require("./itinerary");
+class Ship {
+  constructor(itinerary) {
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.currentPort.addShip(this);
+  }
+
+  setSail() {
+    const itinerary = this.itinerary;
+    const currentPortInd = itinerary.ports.indexOf(this.currentPort);
+    if (currentPortInd === this.itinerary.ports.length - 1) {
+      throw new Error("You have reached the end of the itinerary");
+    }
+    this.currentPort.removeShip(this);
+    this.previousPort = this.currentPort;
+    this.currentPort = null;
```

Comment:
instead of saving the `itinerary` to a variable, you could save the `ports` to a variable, as you only ever use the itinerary to access the ports in this function e.g.
```
const ports = this.itinerary.ports;
const currentPortIndex = ports.indexOf(this.currentPort);

if (currentPortIndex === ports.length - 1) {
  ...
}
...
```